setup/migrate.py

```diff
@@ -191,6 +191,62 @@ def migration_14(env):
 	db = os.path.join(env["STORAGE_ROOT"], 'mail/users.sqlite')
 	shell("check_call", ["sqlite3", db, "CREATE TABLE auto_aliases (id INTEGER PRIMARY KEY AUTOINCREMENT, source TEXT NOT NULL UNIQUE, destination TEXT NOT NULL, permitted_senders TEXT);"])
 
+def migration_15(env):
+	backup_configuration_file = os.path.join(env["STORAGE_ROOT"], 'backup', 'custom.yaml')
+	if os.path.exists(backup_configuration_file):
+		import yaml
+
+		with open(backup_configuration_file, 'r') as f:
+			custom_config = yaml.load(f, Loader=yaml.SafeLoader)
+		if not isinstance(custom_config, dict): raise ValueError()
+
+		# Converting the previous configuration (which was not very clear)
+		# into the new configuration format which also provides
+		# a "type" attribute to distinguish the type of backup.
+		scheme = custom_config["target"].split(":")[0]
+		if scheme == "off":
+			custom_config = {
+				"type": "off"
+			}
+		elif scheme == "file":
+			custom_config = {
+				"type": "local",
+				"min_age_in_days": custom_config["min_age_in_days"]
+			}
+		elif scheme == "rsync":
+			custom_config = {
+				"type": "rsync",
+				"target_url": custom_config["target"],
+				"min_age_in_days": custom_config["min_age_in_days"]
```

Comment:
How sure are we that these keys exist on the dict? Should `or` be used to prevent `KeyError`? ([example](https://stackoverflow.com/a/53263073/1971539) - that might be obvious but I'm not a python guy)

Reply:
Solved with c2636f68f898fd77368350ae2039b08e47de33b3.